arcade/sprite_list/spatial_hash.py

```diff
@@ -1,13 +1,54 @@
+from abc import abstractmethod
 from math import trunc
 from typing import Generic
 
-from arcade.sprite import SpriteType
+from arcade.sprite import SpriteType, SpriteType_co
 from arcade.sprite.base import BasicSprite
 from arcade.types import IPoint, Point
 from arcade.types.rect import Rect
 
 
-class SpatialHash(Generic[SpriteType]):
+class ReadOnlySpatialHash(Generic[SpriteType_co]):
+    """
+    Read-only view of a SpatialHash.
+
+    This is useful when the SpriteType is in covariant position.
+
+    See SpatialHash for more details.
+    """
```

Comment:
Maybe something like this? The wording's less a concern to me than getting the cross-refs solid.
```suggestion
    """A read-only view of a :py:class:`.SpatialHash` which helps preserve safety.

    This works like the read-only views of Python's built-in :py:class:`dict` and other types. As an every-day user, it means
    you can put subtypes of the annotated type into the `SpatialHash` or a `SpriteList`, but not superclasses.
    
    This ensures predicable behavior via type safety in cases where:

    #. A spatial hash is annotated with a specific type
    #. The object could be modified outside the original context to add a broader type
   
    Advanced users who want more information on the specifics should see the comments of :py:class:`~arcade.sprite_list.SpriteList`.
    """
```

Reply:
I took your suggestion but I amended a few things. LMK if you disagree.

Reply:
All good. We can clean it up later. Ty!